### ISSUE TITLE
Upload of pfx certificate file to batch fails for certificates without password

### ISSUE THREAD

**Issue by Zay2k** [Azure/azure-cli#15464]
When trying to upload a pfx certificate file to a batch account a ValidationError occurs. When we change the certificate file to include a password, it works fine. I guess this line marks the file as .cer file not .pfx.

https://github.com/Azure/azure-cli/blob/26fea089cf37a1f6fb4e19f77b120ec0f2477eda/src/azure-cli/azure/cli/command_modules/batch/custom.py#L241

#### Steps to reproduce:
- Create a certificate
    `az keyvault certificate create`
- Download to a .pfx
    `az keyvault secret download`
- Create certificatein batch account
    `az batch certificate create`

#### Error
```
PropertyName: data
Reason: The specified data is an invalid format for a certificate
ValidationError: The value provided for one of the properties in the request body is invalid.
```
#### Debug Output
```
azure.cli.core.util.handle_exception is called with an exception:
Traceback (most recent call last):
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-unpacked-wheel-v8nptj_b\azure\cli\core\commands\__init__.py", line 688, in _run_job
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-unpacked-wheel-v8nptj_b\azure\cli\core\commands\__init__.py", line 325, in __call__
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-unpacked-wheel-v8nptj_b\azure\cli\core\__init__.py", line 784, in default_command_handler
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-unpacked-wheel-s56abhv8\azure\cli\command_modules\batch\custom.py", line 251, in create_certificate
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-unpacked-wheel-fs_zholc\azure\batch\operations\_certificate_operations.py", line 111, in add
azure.batch.models._models_py3.BatchErrorException: Request encountered an exception.
Code: InvalidPropertyValue
Message: {'additional_properties': {}, 'lang': 'en-US', 'value': 'The value provided for one of the properties in the request body is invalid.\nRequestId:93297e5e-250c-49e2-89b7-1a6250796d3a\nTime:2020-10-09T05:49:44.9817887Z'}
PropertyName: data
Reason: The specified data is an invalid format for a certificate
```

**Comment by yungezz** [comment]
hi @bim-msft could you pls help to have a look?

**Comment by yungezz** [comment]
the failed command is `az batch certificate create`, route to service team to have a look.

**Comment by Zay2k** [comment]
This was fixed by PR #15509. Don't know why it isn't linked to this issue.
